src/components/fields/ObjectField.js

```diff
@@ -168,14 +168,33 @@ class ObjectField extends Component {
   }
 
   handleAddClick = schema => () => {
-    const type = schema.additionalProperties.type;
+    let type = schema.additionalProperties.type;
     const newFormData = { ...this.props.formData };
+
+    if (
+      type === undefined &&
```

Comment:
If you have a schema that looks like this:

```
"definitions": {"a": {"type": "number"}},
"additionalProperties": {"$ref": "#/definitions/a", "type": "string"}
```

Then this code will compute the type of each additional property to be `string`.

However, any validation keywords within an object with a `$ref` keyword are not supposed to be applied. So the computed type should actually be `number`.

Reply:
Hm yeah, you're right.

https://deploy-preview-1402--trusting-agnesi-d3347d.netlify.com/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJkZWZpbml0aW9ucyI6eyJhIjp7InR5cGUiOiJudW1iZXIifX0sImFkZGl0aW9uYWxQcm9wZXJ0aWVzIjp7IiRyZWYiOiIjL2RlZmluaXRpb25zL2EiLCJ0eXBlIjoic3RyaW5nIn19LCJ1aVNjaGVtYSI6e319

Reply:
@epicfaace This issue is now resolved. Preview link to see: https://deploy-preview-1402--trusting-agnesi-d3347d.netlify.com/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJkZWZpbml0aW9ucyI6eyJhIjp7InR5cGUiOiJudW1iZXIifX0sImFkZGl0aW9uYWxQcm9wZXJ0aWVzIjp7IiRyZWYiOiIjL2RlZmluaXRpb25zL2EiLCJ0eXBlIjoic3RyaW5nIn19LCJ1aVNjaGVtYSI6e319

Reply:
changed the netlify base url -- [updated preview link](https://deploy-preview-1402--rjsf.netlify.com/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJkZWZpbml0aW9ucyI6eyJhIjp7InR5cGUiOiJudW1iZXIifX0sImFkZGl0aW9uYWxQcm9wZXJ0aWVzIjp7IiRyZWYiOiIjL2RlZmluaXRpb25zL2EiLCJ0eXBlIjoic3RyaW5nIn19LCJ1aVNjaGVtYSI6e319)